src/frameworks/redwoodjs.json

```diff
@@ -0,0 +1,26 @@
+{
+  "id": "redwoodjs",
+  "name": "RedwoodJS",
+  "category": "static_site_generator",
+  "detect": {
+    "configFiles": [
+      "redwood.toml"
+    ]
+  },
+  "dev": {
+    "command": "rw dev",
+    "port": 8910,
+    "pollingStrategies": [
+      {
+        "name": "TCP"
+      }
+    ]
+  },
+  "build": {
+    "command": "yarn rw deploy netlify",
```

Comment:
`yarn` shouldn't be needed. The command is run in the same scope as an npm script, so anything in `node_modules/.bin` is available.

Reply:
@ascorbic Hi. Interesting. We have set the build command that way since day one:

https://github.com/redwoodjs/redwood/blob/fbccb88b31c51258d04d98f5145dcf7c88436168/packages/cli/src/commands/setup/deploy/providers/netlify.js#L6

```
[build]
  command = "yarn rw deploy netlify"
  publish = "web/dist"
  functions = "api/dist/functions"
```

I'll check with the rest of the RedwoodJS Core team and we can try w/out `yarn` and if so, amend the `netlify.toml` we setup when users want to deploy on Netlify.

Reply:
Yes, please do try it first! If you look at all the other frameworks though, you'll see they don't include `yarn` or `npm run` or anything.

Reply:
Ok, I did try just with `netlify dev`:

```
[dev]
  # Note be sure to `netlify link` your site first
  command = "rw dev"
  framework = "#custom"
  targetPort = 8910 # The port for your application server, framework or site generator
  port = 8888 # The port that the netlify dev will be accessible on
```

Without `yarn` and got:

```terminal
◈ Overriding command with setting derived from netlify.toml [dev] block: rw dev
◈ Functions server is listening on 52868
◈ Starting Netlify Dev with #custom
    Error: "rw" could not be found in your PATH. Please make sure that "rw" is installed and available in your PATH
```

I will try with an actual deploy

Reply:
Try it with build rather than dev. `netlify dev` is handled differently from `netlify build`, which uses the same`@netlify/build` package that actual deploys use

Reply:
Well, who knew ...

```terminal
4:44:58 PM: $ rw deploy netlify
4:45:00 PM: yarn run v1.22.4
4:45:00 PM: $ /opt/build/repo/node_modules/.bin/rw build
4:45:03 PM: Generating the Prisma client... [started]
4:45:08 PM: Generating the Prisma client... [completed]
4:45:08 PM: [20:45:08] Cleaning "web"... (./web/dist/) [started]
4:45:08 PM: [20:45:08] Cleaning "web"... (./web/dist/) [completed]
4:45:08 PM: [20:45:08] Building "api"... [started]
4:45:11 PM: [20:45:11] Building "api"... [completed]
4:45:11 PM: [20:45:11] Building "web"... [started]
4:45:44 PM: [20:45:44] Building "web"... [completed]
4:45:44 PM: [20:45:44] Prerendering "web"... [started]
4:45:50 PM: [20:45:50] Prerendering "web"... [completed]
4:45:50 PM: Done in 49.71s.
```

Deploying on Netlify didn't need yarn.

I'll circle back with the Core Team and discuss if we want to remove that and change the deploy setup and docs.

Reply:
Confirmed that `netlify build` locally also did not need `yarn`.

Reply:
Great stuff! Thanks for submitting this. You'll also want to add it as an export in `main.js`.

Reply:
Ah, I missed that. I see now ... I didn't catch that browsing the code or the instructions in the readme:

```
Add or update a framework
Each framework is a JSON file in the /src/frameworks/ directory. 
```

Reply:
@ascorbic Might you tell me what the 

```
[build]
  command = "rw deploy netlify"
```

Is for in this file?

Is it used for detection or just docs?

The team's preference would be to keep it `yarn rw deploy netlify` but I can have them make their case.

Reply:
The reason yarn is specifically being used:

> A Redwood application is split into two parts: a frontend and a backend. This is represented as two node projects within a single monorepo. We use Yarn to make it easy to operate across both projects while keeping them in a single Git repository.

https://redwoodjs.com/docs/introduction.html#how-it-works

Reply:
> yarn shouldn't be needed. The command is run in the same scope as an npm script, so anything in node_modules/.bin is available.

@ascorbic just so I'm clear, my interpretation here is that you're making a recommendation in order to simplify the command. Is that correct? 

**Quick Redwood DX background** --> We do not take the path of managing a separate CLI package via global installation. Instead, we include CLI within our core framework package. This means Redwood Devs are _always_ using `yarn redwood` (or `rw`) to run commands.

Removing it here seems mechanically viable. However, it would be an exception to the expected usage for any dev. For consistency (and given I understand correctly), my recommendation is to stick with `yarn rw ...`. If not, we should probably document that it's not needed to avoid confusion.

Reply:
When a framework is detected, you don't need to put a build command in your netlify.toml: it will instead run the command specified in this JSON file. This is also the command that is pre-populated in the UI when creating a new site with this framework. Once this is merged, somebody will be able to deploy a RedwoodJS repo without a netlify.toml, and it will know the correct command to run to build it.

It's not a problem if you want to keep it as `yarn rw...`, but it does mean it will need to spawn an extra unnecessary process to do the build.

Reply:
@thedavidprice Yes, it's just a recommendation. This is mostly used internally, so it's about whether it calls `execa("rw deploy", { preferLocal: true })` or `execa("yarn rw deploy", { preferLocal: true })`

Reply:
> it will instead run the command specified in this JSON file

If someone does supply a build command in `netlify.toml`, will that override the "default" command specified in the JSON file?

Reply:
@ascorbic Thanks for the explanation. How about this....

* let me change it back top use `yarn` so we keep to the Redwood conventions
* we can then test that the detection is working properly
* I'll circle back w/ the team and discuss changing the command 
* also do some build tests to see if build times improve w/out yarn (that would be a point in removing yarn's favor)

Sound good? I'll make that change back now.

Reply:
@thedavidprice
> If someone does supply a build command in `netlify.toml`, will that override the "default" command specified in the JSON file?

Yes, correct

@dthyresson sounds good

Reply:
@dthyresson I'll connect with you via Slack, but my recommendation at this point is:
- put everything in the JSON file that we can
- in the JSON file, use `rw deploy netlify` (without `yarn` command)
- use those as defaults
- modify the `rw setup deploy netlify` template to comment out the options — use the file to give instruction about how to add overrides (if needed) and which config options are available.

Reply:
@ascorbic Sorry it looks like my builds are failing. I didn't see any instructions in the Readme to do a build or run tests, so I apologize for not trying that before adding the PR.

I'm trying to figure out how to build and run the tests locally ...

```
  npm help
framework-info add-redwoodjs-detection % npm run build

> @netlify/framework-info@4.1.0 build
> run-s build:*

sh: run-s: command not found
```

So I'll try to sort this out and add any fixes. I looked at the Blitz PR that was recently done and only one file was added, so I patterned the commit to that.

Reply:
@ascorbic Here's the default `netlify.toml` we add to a project for Netlify hosting (apologies if you've already seen this):
https://github.com/redwoodjs/redwood/blob/e6c941aea36433ebb790c75c62a0a0dfc9edb998/packages/cli/src/commands/setup/deploy/providers/netlify.js#L5-L16

```
[build]
  command = "yarn rw deploy netlify"
  publish = "web/dist"
  functions = "api/dist/functions"

[dev]  
  command = "yarn rw dev"

[[redirects]]  
  from = "/*"  
  to = "/index.html"  
  status = 200
```

Can the [[redirects]] be added in the JSON as defaults as well?

Reply:
Ok I was able to fix the build and test, though I was getting a local Cypress error in e2e.

>   2) Detects frameworks
       Next.js site framework detected on load - vanilla JS site:
     AssertionError: Timed out retrying after 4000ms: Expected to find content: '"name":"Next.js"' but never did.
      at Context.eval (http://localhost:8080/__cypress/tests?p=cypress/integration/e2e.js:109:8)

I'll see how it runs in CI.

I may also add some instructions in README to lett others know how to build and test locally.

I had to add 

```
    "npmDependencies": [
      "@redwoodjs/core"
    ],
    "excludedNpmDependencies": [],
```

Even though I thought just a config entry would be enough for detection -- but the test failed.

`npmDependencies` could not be empty.

Reply:
FYI

```
npmDependencies
Type: string[]

Framework's npm packages. Any project with one of those packages in their package.json (dependencies or devDependencies) will be considered as using the framework.

If empty, this is ignored.

excludedNpmDependencies
Type: string[]

Inverse of npmDependencies. If any project is using one of those packages, it will not be considered as using the framework.

If empty, this is ignored.
```

That's why I left `npmDependencies` off the config initially, but then when i left it empty `[]` my tests failed.

Reply:
Ah, my mistake. The `CONTRIBUTING` files has the build and test info. I forgot to look there.

Reply:
I'm afraid no redirects in here right now, but we plan to have an API for build plugins to add them, so that would be a point where we could add one